python/pylibcudf/pyproject.toml

```diff
@@ -101,6 +102,8 @@ matrix-entry = "cuda_suffixed=true"
 requires = [
     "cmake>=3.26.4,!=3.30.0",
     "cython>=3.0.3",
+    "libcudf==24.10.*,>=0.0.0a0",
+    "librmm==24.10.*,>=0.0.0a0",
```

Comment:
Without this `librmm` build dependency (for both `pylibcudf` and `cudf`), I saw builds of those libraries fail like this:

```text
  CMake Error at /tmp/pip-build-env-7u64pfti/normal/lib/python3.11/site-packages/cmake/data/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
    By not providing "Findnvtx3.cmake" in CMAKE_MODULE_PATH this project has
    asked CMake to find a package configuration file provided by "nvtx3", but
    CMake did not find one.

    Could not find a package configuration file provided by "nvtx3" with any of
    the following names:

      nvtx3Config.cmake
      nvtx3-config.cmake

    Add the installation prefix of "nvtx3" to CMAKE_PREFIX_PATH or set
    "nvtx3_DIR" to a directory containing one of the above files.  If "nvtx3"
    provides a separate development package or SDK, be sure it has been
    installed.
  Call Stack (most recent call first):
    /tmp/pip-build-env-7u64pfti/normal/lib/python3.11/site-packages/libcudf/lib64/cmake/cudf/cudf-dependencies.cmake:24 (find_dependency)
    /tmp/pip-build-env-7u64pfti/normal/lib/python3.11/site-packages/libcudf/lib64/cmake/cudf/cudf-config.cmake:90 (include)
    CMakeLists.txt:27 (find_package)
```

`librmm` vendors `nvtx3-config.cmake` and the `nvtx` headers... is it ok to rely on that here?

```shell
pip download \
    --no-deps \
    --extra-index-url https://pypi.anaconda.org/rapidsai-wheels-nightly/simple/ \
    'librmm-cu12==24.10.*,>=0.0.0a0'

unzip -l ./librmm_cu12*.whl \
| grep -i nvtx
```

```text
    50660  2024-08-20 23:50   librmm/include/nvtx3/nvToolsExt.h
...
    21666  2024-08-20 23:50   librmm/include/nvtx3/nvtxDetail/nvtxImpl.h
...
     4788  2024-08-20 23:50   librmm/include/rapids/cub/detail/nvtx.cuh
   105816  2024-08-20 23:50   librmm/include/rapids/cub/detail/nvtx3.hpp
     1610  2024-08-20 23:49   librmm/include/rmm/detail/nvtx/ranges.hpp
     3675  2024-08-20 23:50   librmm/lib64/cmake/nvtx3/nvtx3-config-version.cmake
     4937  2024-08-20 23:50   librmm/lib64/cmake/nvtx3/nvtx3-config.cmake
     4357  2024-08-20 23:50   librmm/lib64/cmake/nvtx3/nvtx3-targets.cmake
```

Reply:
I don't know the best approach here. For now I'm okay with this. This is probably a similar kind of problem as conda packages including CPM dependencies, which leads to clobbering, but we won't have clobbering with the wheels since they all have their own install paths.

Reply:
So basically what's happening here is that the libcudf wheel does not include nvtx because when building it nvtx is found within the librmm wheel, right? But even aside from nvtx, won't you need librmm available for compilation? Any cudf Cython file that includes libcudf headers that in turn include librmm headers will require those librmm headers to be available at build time, and we don't want to CPM download those. IOW https://github.com/rapidsai/build-planning/issues/92.

Reply:
> *...the libcudf wheel does not include nvtx because when building it nvtx is found within the librmm wheel, right?*

Yes exactly.

> *But even aside from nvtx, won't you need librmm available for compilation?*

Yeah I think the way you've described this is exactly correct, and described well in https://github.com/rapidsai/build-planning/issues/92. Everything in the chain of build dependencies (`libcudf -> pylibcudf -> cudf`) needs a build dependency on `librmm` because none of them take a *runtime* dependency on `librmm`.

Which I recall was an intentional choice as part of the discussion around https://github.com/rapidsai/build-planning/issues/92.

So given that we're going to want the `librmm` build dependency *anyway* I guess there's nothing to worry about here. Kind of figured, but wanted to validate that assumption. Thank you!

Reply:
Yup exactly the nvtx piece is just one manifestation of the broader need for the librmm dependency being required.